src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -456,36 +456,69 @@ private async Task<ExportJobOutcome> UpdateExportJobAsync(ExportJobRecord export
             // 2. There is no continuation token but the page is 0, which means it's the initial export.
             while (progress.ContinuationToken != null || progress.Page == 0)
             {
-                SearchResult searchResult = null;
+                List<SearchResultEntry> searchResultEntries = new();
+                string continuationToken = null;
 
                 // Search and process the results.
                 switch (_exportJobRecord.ExportType)
                 {
                     case ExportJobType.All:
+                        using (IScoped<ISearchService> searchService = _searchServiceFactory())
+                        {
+                            // Fetch all the expected results
+                            do
+                            {
+                                var allSearchResult = await searchService.Value.SearchAsync(
+                                    resourceType: resourceType,
+                                    queryParametersList,
+                                    cancellationToken,
+                                    true);
+
+                                searchResultEntries.AddRange(allSearchResult.Results);
+
+                                queryParametersList.RemoveAll(x => x.Item1 == KnownQueryParameterNames.ContinuationToken);
+                                continuationToken = allSearchResult.ContinuationToken;
+
+                                if (continuationToken is not null)
+                                {
+                                    queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.ContinuationToken, ContinuationTokenConverter.Encode(continuationToken)));
+                                }
+                            }
+                            while (searchResultEntries.Count < _exportJobRecord.MaximumNumberOfResourcesPerQuery || continuationToken is not null);
```

Comment:
This should be an AND, no OR.

Reply:
Changed the implementation, code was deleted.